packages/tools/devtools/devtools-view/src/components/data-visualization/EditableView.tsx

```diff
@@ -0,0 +1,384 @@
+ */
+interface EditableBooleanComponentProps {
+	node: FluidObjectValueNode;
+	submitChange: (data: Serializable<unknown> | Primitive) => void;
+}
+
+/**
+ * Component which allows for boolean editing
+ */
+function EditableBooleanComponent(props: EditableBooleanComponentProps): React.ReactElement {
+	const { node, submitChange } = props;
+
+	/**
+	 * Auxiliary function to make boolean editing buttons
+	 * @param isTrue - Determines if it should be a true or false button
+	 * @returns - 2 buttons, one for setting value to false and one for true
+	 */
+	function boolButton(isTrue: boolean): React.ReactElement {
+		return (
+			<Button
+				size="small"
+				disabled={isTrue === node.value}
+				onClick={(event): void => {
+					event.preventDefault();
+					event.stopPropagation();
+					submitChange(isTrue);
+				}}
+			>
+				{String(isTrue).charAt(0).toUpperCase() + String(isTrue).slice(1)}
+			</Button>
+		);
+	}
+	return (
+		<>
+			{boolButton(true)}
+			{boolButton(false)}
+		</>
+	);
+}
+
+/**
+ * Input to {@link EditTextComponent}
+ */
+interface EditableTextComponentProps {
+	node: FluidObjectValueNode;
+	setEditing: React.Dispatch<React.SetStateAction<boolean>>;
+	submitChange: (data: Serializable<unknown> | Primitive) => void;
+}
+
+/**
+ * Component which allows for text editing
+ */
+function EditableTextComponent(props: EditableTextComponentProps): React.ReactElement {
+	const { node, setEditing, submitChange } = props;
+	const [localData, setLocalData] = React.useState<string>(
+		typeof node.value !== "string" ? "" : node.value,
+	);
+
+	/**
+	 * Provides comfirming and canceling an edit functionality
+	 */
+	const onKeyDown = React.useCallback(
+		(event: React.KeyboardEvent<HTMLInputElement>) => {
+			const key = event.key;
+
+			if (key === "Enter") {
+				event.currentTarget.blur();
+				submitChange(localData);
+			}
+
+			if (key === "Escape") {
+				event.currentTarget.blur();
+			}
+		},
+		[localData, submitChange],
+	);
+
+	/**
+	 * Sets the local data to what the user wrote in the component
+	 */
+	function onChange(_ev: React.ChangeEvent<HTMLInputElement>, data: InputOnChangeData): void {
+		setLocalData(data.value);
+	}
+
+	/**
+	 * Sets {@link editing} to false when the compoenent is focused
+	 */
+	const onFocus = React.useCallback(() => {
+		setEditing(true);
+	}, [setEditing]);
+
+	return (
+		<>
+			<Input
+				size="small"
+				appearance="underline"
+				contentEditable={true}
+				value={localData}
+				onFocus={onFocus}
+				onKeyDown={onKeyDown}
+				onChange={onChange}
+				type="text"
+			/>
+		</>
+	);
+}
+
+/**
+ * Input to {@link EditTextComponent}
+ */
+interface EditableNumberComponentProps {
+	node: FluidObjectValueNode;
+	setEditing: React.Dispatch<React.SetStateAction<boolean>>;
+	submitChange: (data: Serializable<unknown> | Primitive) => void;
+}
+
+/**
+ * Component which allows for number editing
+ */
+function EditableNumberComponent(props: EditableNumberComponentProps): React.ReactElement {
```

Comment:
I think we can probably share a lot of code between this component and the Text one. Probably worth combining them and adding the necessary props.

Reply:
You suggested we combine all the props to only have one "EditableComponentProps", using that even the button would need to recieve dud information if we add the necesary props to distinguish the inputs. Should I do it anyway? What are your thoughts?